domain/application/service/application.go

```diff
@@ -664,8 +666,20 @@ func (s *Service) DestroyUnit(ctx context.Context, unitName coreunit.Name) error
 // If the unit is not found, an error satisfying [applicationerrors.UnitNotFound]
 // is returned.
 func (s *Service) EnsureUnitDead(ctx context.Context, unitName coreunit.Name, leadershipRevoker leadership.Revoker) error {
-	err := s.st.RunAtomic(ctx, func(ctx domain.AtomicContext) error {
-		return s.ensureUnitDead(ctx, unitName)
+	unitLife, err := s.st.GetUnitLife(ctx, unitName)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if unitLife == life.Dead {
+		return nil
+	}
+	err = s.st.RunAtomic(ctx, func(ctx domain.AtomicContext) error {
+		// TODO(units) - check for subordinates and storage attachments
```

Comment:
Can you please add (or make sure it exists) a jira task for this under the unit epic?